### ISSUE TITLE
Crash when running lab01.wad

### ISSUE THREAD

**Issue by munday1** [chocolate-doom/chocolate-doom#994]
### Background

Version of Chocolate Doom: Commit 0b50fbde6a49250d4372a3255b1e1fd451326abe (Master branch)

Operating System and version:Windows XP SP2

Game: Doom 2 V 1.9

Any loaded WADs and mods (please include full command line):chocolate-doom -file lab01.wad (https://www.doomworld.com/files/file/13295-lab01wad)

### Bug description

Observed behavior:When running the PWAD with sound (regardless of music on or off),after the first castle once exiting the cave,turn to the left tunnel and press the switch at the end of the corridor, the game crashes.

Expected behavior:The switch should've triggered the door to open.

**Comment by fabiangreffrath** [comment]
I just tried to reproduce this, but I have no idea where on this map the switch might be. Could you probably provide a savegame or a screenshot?

**Comment by munday1** [comment]
I would,but my browser isn't supported by Github.

**Comment by munday1** [comment]
https://drive.google.com/file/d/1orwV3eHHa85b9zSga-7F7pk3Y1QSZM9c/view?usp=sharing
Here,

**Comment by munday1** [comment]
Wait,don't view it.(Not correct)

**Comment by munday1** [comment]
https://drive.google.com/file/d/1ctdTSY8lYZ3EJELnbyzhjCBwII6lr48P/view?usp=sharing (Correct)
https://drive.google.com/file/d/1G6PNOo0ehE5XK4TEr5TfrBNYeoNw_au3/view?usp=sharing (Automap where I was standing.)

**Comment by fabiangreffrath** [comment]
Thank you, but, sorry, I still cannot reproduce the crash.

**Comment by JNechaevsky** [comment]
I can confirm this on Windows 10 (Pro, x64):
- https://youtu.be/_SXU0us1WHw

Tested with latest dev build (20180221-win32). I also can confirm that this crash is not happening with `-nosfx` command line parameter and/or with zero SFX volume.

I would like to test it on Debian, but need to remember how use gdb first. :confused:

**Comment by fabiangreffrath** [comment]
Interesting! Would you mind sharing the savegame?

> I would like to test it on Debian, but need to remember how use gdb first.

`gdb --args chocolate-doom -file lab01.wad ` and then `r` once gdb started.

**Comment by JNechaevsky** [comment]
Sure thing! Here you go, but after loading this game there is no crash - I've made it just for sure. More of that, even after loading this save and then starting a new game, crash will not happen again!
Really odd...

- https://yadi.sk/d/YWJB4ykX3Sf76r

> gdb --args chocolate-doom -file lab01.wad and then r once gdb started.

Oh yes, that it, thanks! Will test shortly.

**Comment by JNechaevsky** [comment]
Hm. Got no crash, but have noted some kind of performance drop after pressing that switch. Video record:

- https://youtu.be/iY8Tmhhh3qs

FPS drop is happening while door is opened, probably that's a key... I need to inspect this map in Doom Builder. Sorry for awful sound, I still have SDL 2.0.6.

**Comment by JNechaevsky** [comment]
Done inspecting, here's my results:

- Doom Builder is not showing any errors in that area
- Rebuiling NODES is not helping (even by adding new sector somewhere and resaving)

Hovewer, one thing attracts my attention. These columns near the problematic door are not single-sided and having some odd architecture. Probably it was a mapping hack, or, well, _violation_ of principles of map building:

![](http://screenshot.ru/upload/images/2018/02/21/hmm9a780.png)

Converting two these columns to a single-sided lines fixes this issue:

![](http://screenshot.ru/upload/images/2018/02/21/hmm26da58.png)

But what's the real reason of crash/framedrop then? Probably some calculations are going crazy because of this structure - this is explaining crash after turning around and looking to this opened door. But why crash is not happening with turned off sound - it's a question of the day, I even can't imagine.

**Comment by JNechaevsky** [comment]
Tested this case with original Doom II v 1.9 for DOS. There is no crash, not framedrop:

- https://youtu.be/L_g8CJAhXv0

**Comment by JNechaevsky** [comment]
Chocolate Doom 2.3.0 is also crashing, without any crash information in `stderr.txt` and `stdout.txt`.

**Comment by fabiangreffrath** [comment]
Is the texture in question composed of more than one patch? If the answer is "yes" and changing the lines to be single-sided helps fixing the issue, this sounds a lot like the Medusa bug to me.

**Comment by munday1** [comment]
Then it should've crashed DOOM2.EXE as well, right?

**Comment by fabiangreffrath** [comment]
Not necessarily, it depends on the memory layout at run-time.

**Comment by jmtd** [comment]
Those pillars are ROCK2, which appears to be a single-texture patch of AROCK2. But the switch itself is a 2s linedef and that's 2 patches, WALL47_2 and WALL42_1. So this is almost certainly Medusa, good catch!

**Comment by JNechaevsky** [comment]
I've also have to agreed that it must be Medusa-related, but not everything that simple as it seems. :/

Jon is absolutely right, green rocky texture `ROCK2` is a single patched, `SW1SATYR` is double patched.

But a question then - why Medusa dissapears when I'm converting `ROCK2` pillars to single-sided linedefs? Those pillars must not be relater to Medusa, right? To check it, I will replace `SW1SATYR` to single-patched `ROCK2`, save and try to run:

- Poof! https://youtu.be/l1CsjmtR4v8

One more test. Medusa is fixed in Crispy and my fork, but in my version for DOS, despite of many significant additions and enforcements from Crispy, I was not lucky with fixing Medusa (too much difference in the code). Let's try it just for sure:

- No framedrops, no visible Medusa: https://youtu.be/_PlC8pINfrM

Could be there something significantly different between DOS-Windows-Linux in calculations? But why even vanilla DOS versions working fine in this case? And again, why disabled sound fixing this?

My only suggestion now is... Pfff, how should I say... Something going crazy in the engine while trying to find nearest height for the sector with opening door. Or maybe I just don't have a perfect understanding what actually Medusa is?

**Comment by JNechaevsky** [comment]
Small update. Made another experiment by leaving problematic door "slightly opened" ([screenshot](http://screenshot.ru/upload/images/2018/02/22/hmm38183e.png)) and trying to run this map:

- Crash! https://youtu.be/IDoWo12C0ac

As you can see, I've even haven't touched opening switch. So, my theory about height calculations is  completely wrong _(luckily!)_, it have to be something like "Invisible Medusa" + something else? Wish I understand this case more deeper...

**Comment by AXDOOMER** [comment]
Do I understand correctly: It's crashing only on Windows and not on Linux?

I could run it in Visual Studio in order to get the stack trace.

**Comment by AXDOOMER** [comment]
>  I also can confirm that this crash is not happening with -nosfx command line parameter and/or with zero SFX volume.

I'm sure we have seen that in the past.

**Comment by JNechaevsky** [comment]
> Do I understand correctly: It's crashing only on Windows and not on Linux?

Yep, exactly. Vanilla DOS version does not suffering this problem at all.

**Comment by AXDOOMER** [comment]
Hi guys,

Here's the stack trace when the program crashes:
![stacktrace](https://user-images.githubusercontent.com/6194072/36653414-0b4e6910-1a83-11e8-8513-b80e8cd32550.png)

**Comment by AXDOOMER** [comment]
The reason of the crash is a segmentation fault. 

![where it crashes](https://user-images.githubusercontent.com/6194072/36653679-d51c94dc-1a84-11e8-9a0c-81bac2df4a7b.png)
